core/src/main/java/org/apache/accumulo/core/util/threads/ThreadPools.java

```diff
@@ -232,180 +237,257 @@ private ThreadPools(UncaughtExceptionHandler ueh) {
+        if (emitThreadPoolMetrics) {
+          builder.enableThreadPoolMetrics();
+        }
+        return builder.build();
+
       default:
-        throw new RuntimeException("Unhandled thread pool property: " + p);
+        throw new IllegalArgumentException("Unhandled thread pool property: " + p);
     }
   }
 
-  /**
-   * Create a named thread pool
-   *
-   * @param numThreads number of threads
-   * @param name thread pool name
-   * @param emitThreadPoolMetrics When set to true will emit metrics and register the metrics in a
-   *        static registry. After the thread pool is deleted, there will still be metrics objects
-   *        related to it in the static registry. There is no way to clean these left over objects
-   *        up therefore its recommended that this option only be set true for long lived thread
-   *        pools. Creating lots of short lived thread pools and registering them can lead to out of
-   *        memory errors over long time periods.
-   * @return ThreadPoolExecutor
-   */
-  public ThreadPoolExecutor createFixedThreadPool(int numThreads, final String name,
-      boolean emitThreadPoolMetrics) {
-    return createFixedThreadPool(numThreads, DEFAULT_TIMEOUT_MILLISECS, MILLISECONDS, name,
-        emitThreadPoolMetrics);
+  public PoolBuilder getPoolBuilder() {
+    return new PoolBuilder();
   }
 
-  /**
-   * Create a named thread pool
-   *
-   * @param numThreads number of threads
-   * @param name thread pool name
-   * @param queue queue to use for tasks
-   * @param emitThreadPoolMetrics When set to true will emit metrics and register the metrics in a
-   *        static registry. After the thread pool is deleted, there will still be metrics objects
-   *        related to it in the static registry. There is no way to clean these left over objects
-   *        up therefore its recommended that this option only be set true for long lived thread
-   *        pools. Creating lots of short lived thread pools and registering them can lead to out of
-   *        memory errors over long time periods.
-   * @return ThreadPoolExecutor
-   */
-  public ThreadPoolExecutor createFixedThreadPool(int numThreads, final String name,
-      BlockingQueue<Runnable> queue, boolean emitThreadPoolMetrics) {
-    return createThreadPool(numThreads, numThreads, DEFAULT_TIMEOUT_MILLISECS, MILLISECONDS, name,
-        queue, emitThreadPoolMetrics);
-  }
+  public class PoolBuilder {
+    int coreThreads = 0;
+    int maxThreads = -1;
+    long timeOut = DEFAULT_TIMEOUT_MILLISECS;
+    TimeUnit units = MILLISECONDS;
+    String name = null;
+    BlockingQueue<Runnable> queue = new LinkedBlockingQueue<>();
+    OptionalInt priority = OptionalInt.empty();
+    boolean emitThreadPoolMetrics = false;
+
+    /**
+     * A fluent-style build to create a ThreadPoolExecutor
+     */
+    PoolBuilder() {}
+
+    public ThreadPoolExecutor build() {
+      Objects.requireNonNull(name, "a thread name must be provided");
```

Comment:
If name is required, why not expect it in the constructor and have the callers pass it in the call to getPoolBuilder?

Reply:
Fixed in 1dba52a521